deploy/determined_deploy/gcp/terraform/modules/compute/main.tf

```diff
@@ -46,6 +51,8 @@ resource "google_compute_instance" "master_instance" {
       max_agent_starting_period: ${var.max_agent_starting_period}
       provider: gcp
       name_prefix: det-dynamic-agent-${var.unique_id}-${var.det_version_key}-
+      label_key: managed-by
+      label_value: det-master-${var.unique_id}-${var.det_version_key}
```

Comment:
GCP label value has a max length of 64 chars. I think this could exceed tha. Maybe put a max length of 35 or 40 on the `cluster_id` argument in `cli.py`?

It looks like you can pass a lambda in to the type field of argparse to add validation easily: https://stackoverflow.com/questions/25470844/specify-format-for-input-arguments-argparse-python

Reply:
IMO we should handle validation in a separate, broader, non blocking JIRA, because it's not a new problem and friendly validation potential in det-deploy goes well beyond this one case

Reply:
Fair enough. But if you change line 24-30 in `gcp/cli.py` to 
```
    def validate_cluster_id(cluster_id):
        if len(cluster_id) > 35:
            raise argparse.ArgumentTypeError("cluster-id must be 35 characters or shorter")

    required_named.add_argument(
        "--cluster-id",
        type=validate_cluster_id,
        default=None,
        required=True,
        help="unique identifier to name and tag resources",
    )
```
it's one less thing we have to fix in the future. PR is approved either way.

Reply:
thanks, added this check